aten/src/ATen/core/TensorImpl.h

```diff
@@ -19,11 +19,29 @@ struct Storage;
 struct Tensor;
 } // namespace at
 
+namespace at {
+struct TensorImplOptions {
+  bool is_variable_;
+  bool has_storage_;
+  bool has_strides_;
+  TensorImplOptions(bool is_variable)
```

Comment:
on second thought, I think we should get rid of this constructor and make people specify everything.  It's too easy to make a mistake (see the variable issue below).